iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/Replace3TwoWindingsTransformersByThreeWindingsTransformersTest.java

```diff
@@ -0,0 +1,440 @@
+
+    @BeforeEach
+    public void setUp() {
+        Network expectedNetwork = createSetUpNetwork();
+        assertEquals(3, expectedNetwork.getTwoWindingsTransformerCount());
+        t2w1 = expectedNetwork.getTwoWindingsTransformer("3WT-Leg1");
+        t2w2 = expectedNetwork.getTwoWindingsTransformer("3WT-Leg2");
+        t2w3 = expectedNetwork.getTwoWindingsTransformer("3WT-Leg3");
+        network = createSetUpNetwork();
+    }
+
+    private static Network createSetUpNetwork() {
+        Network network = ThreeWindingsTransformerNetworkFactory.create();
+        ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers replace = new ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers();
+        replace.apply(network);
+        return network;
+    }
+
+    @Test
+    void replaceBasicTest() {
+        assertEquals(4, network.getVoltageLevelCount());
+        assertEquals(3, network.getTwoWindingsTransformerCount());
+        assertEquals(0, network.getThreeWindingsTransformerCount());
+
+        Replace3TwoWindingsTransformersByThreeWindingsTransformers replace = new Replace3TwoWindingsTransformersByThreeWindingsTransformers();
+        replace.apply(network);
+
+        assertEquals(3, network.getVoltageLevelCount());
+        assertEquals(0, network.getTwoWindingsTransformerCount());
+        assertEquals(1, network.getThreeWindingsTransformerCount());
+    }
+
+    @Test
+    void replaceRatioTapChangerTest() {
+        assertNull(t2w1.getRatioTapChanger());
+        assertNotNull(t2w2.getRatioTapChanger());
+        assertNotNull(t2w3.getRatioTapChanger());
+
+        Replace3TwoWindingsTransformersByThreeWindingsTransformers replace = new Replace3TwoWindingsTransformersByThreeWindingsTransformers();
+        replace.apply(network);
+        ThreeWindingsTransformer t3w = network.getThreeWindingsTransformer("3WT-Leg1-3WT-Leg2-3WT-Leg3");
+
+        assertNull(t3w.getLeg1().getRatioTapChanger());
+        assertNotNull(t3w.getLeg2().getRatioTapChanger());
+        assertNotNull(t3w.getLeg3().getRatioTapChanger());
+
+        assertTrue(compareRatioTapChanger(t2w2.getRatioTapChanger(), t3w.getLeg2().getRatioTapChanger()));
+        assertTrue(compareRatioTapChanger(t2w3.getRatioTapChanger(), t3w.getLeg3().getRatioTapChanger()));
+    }
+
+    @Test
+    void replacePhaseTapChangerTest() {
+        modifyNetworkForPhaseTapChangerTest();
+
+        assertNotNull(t2w1.getPhaseTapChanger());
+        assertNull(t2w2.getPhaseTapChanger());
+        assertNull(t2w3.getPhaseTapChanger());
+
+        Replace3TwoWindingsTransformersByThreeWindingsTransformers replace = new Replace3TwoWindingsTransformersByThreeWindingsTransformers();
+        replace.apply(network);
+        ThreeWindingsTransformer t3w = network.getThreeWindingsTransformer("3WT-Leg1-3WT-Leg2-3WT-Leg3");
+
+        assertNotNull(t2w1.getPhaseTapChanger());
+        assertNull(t2w2.getPhaseTapChanger());
+        assertNull(t2w3.getPhaseTapChanger());
```

Comment:
Don't you want to test t3w.getLegX() instead of t2wX ?

Reply:
You are right. Done